src/main/java/com/paneedah/weaponlib/EntityProjectile.java

```diff
@@ -85,8 +86,8 @@ public EntityProjectile(World world, EntityLivingBase thrower, float velocity, f
 
     public void setPositionAndDirection() {
 
-        this.setLocationAndAngles(thrower.posX, thrower.posY + (double) thrower.getEyeHeight(),
-                thrower.posZ, compatibility.getCompatibleAimingRotationYaw(thrower), thrower.rotationPitch);
+        this.setLocationAndAngles(thrower.posX - 0.16 , thrower.posY + (double) thrower.getEyeHeight(),
+                thrower.posZ - 0.16 , compatibility.getCompatibleAimingRotationYaw(thrower), thrower.rotationPitch);
```

Comment:
That's wrong, the problem is not that it fires off center, the problem is that it always fires off center, when the weapon is on your right hand it is correct and logical for the gun to shoot from there, but when it is right in front of you and you are aiming now it is not correct, so the centering should only be done when the player is aiming down sight.

Reply:
I understand. Should the position generated by the bullet be based on the player's perspective? It's just that it hasn't been implemented yet

Reply:
I have no idea you can look into it or just wait until we fix it

Reply:
How should I obtain if the player is targeting

Reply:
I know now

Reply:
> I have no idea you can look into it or just wait until we fix it

I fixed it

Reply:
https://streamable.com/2ye99m